File: Tokens/ArtistToken.cs
using iTunesLib;

namespace iTunesRichPresence_Rewrite.Tokens {
    public class ArtistToken : IToken {
        public string DisplayName => "Artist";
        public string Token => "%artist";
        public string GetText(IiTunes iTunes) {
            return iTunes.CurrentTrack.Artist;
        }
    }
}

File: src/main/templating/tokens/ArtistToken.ts
import { Token } from '../types';
import { CurrentSong } from '../../itunes/types';

export default class ArtistToken implements Token {
    token = 'artist';
    displayName = 'Artist';

    render(currentSong: CurrentSong): string {
        return currentSong.Artist;
    }
}
